pkg/object/azure.go

```diff
@@ -104,12 +104,16 @@ func (b *wasb) List(prefix, marker string, limit int64) ([]Object, error) {
 
 	limit32 := int32(limit)
 	pager := b.container.ListBlobsFlat(&azblob.ContainerListBlobFlatSegmentOptions{Prefix: &prefix, Marker: &marker, Maxresults: &(limit32)})
+	if pager.Err() != nil {
+		return nil, pager.Err()
+	}
+	var n int
 	if pager.NextPage(ctx) {
 		b.marker = *pager.PageResponse().NextMarker
+		n = len(pager.PageResponse().Segment.BlobItems)
```

Comment:
n should be the number of items in current page, n could be > 0 if even NextPage() is false